### ISSUE TITLE
Media change during P2P leads to JVB routing stream it's not supposed to (bandwidth leak)

### ISSUE THREAD

**Issue by jusquiame** [jitsi/jitsi-meet#10754]
## Description 
While in a 1to1 conference one or both navigating on Chrome and P2P working, switching media (camera, screen-share, plugging a camera when none was) leads to the JVB receiving and/or sending the new media stream. I think it's a merely redundant artefact, see remark (1) below.

This doesn't happen when both attendees use Firefox.

## Current behavior 
JVB gets loaded while in P2P connection

## Expected Behavior
switching media while in P2P doesn't load the videobridge. 

## Possible Solution
????

## Possible ugly patch to save those whose P2P is 100% use and bandwidth usage vital
renegotiating P2P connection from scratch upon any streamed media change. 

## Steps to reproduce
Note : I worked on a dev instance on which I was alone, thus the following method. 
Another user reproduced the problem on last unstable, observing transceiver/endpointConnectionStats in debug mode.

- On the instance with JVB you're the only one to use, start nload or any traffic monitoring tool you like.

- Start a 1to1 conference on Chrome, makes sure P2P works properly. You should see no or near zero traffic on your instance, but a short burst upon connexions of course. 

- Start screen sharing. Traffic rises to correlation with screen-sharing stream 

- Stop screen sharing, traffic on the bridge is correlated to resumed camera video stream. 

- mute/unmute the camera and observe it stops/resumes the traffic on JVB.

##  Additional remarks : 

 1. Sometime, the load on JVB follows the following pattern :
![half_relay_nload_jitsi2](https://user-images.githubusercontent.com/7765660/145607950-507965be-2cb5-44b3-9325-301569406109.png)
After outgoing traffic went back to near zero, the conference was still live, with neither interruption in audio or video, nor change in quality or framerate. But incoming traffic on server stayed correlated to video stream. Suggests that this is a ghost stream, routed for nothing. 

2. I think this behavior is linked to recent (and mandatory) switch to unified plan. 

3. As mentioned, Chrome seems to be part of the problem as when both users are on FF - that has always used Unified Plan - the problem is not observed.

# Environment details

- Navigating with Chrome 96

observed on : 
- last unstable
- 2 different self-hosted instances having slightly different settings but common versions :
    Ubuntu                                                                        18.04
    jitsi-meet				 	                                2.0.6689-1, 
    prosody, tokens, turnserver, web, web-config 	        1.0.5638-1                                      
    videobridge2                    	 	                                2.1-592-g1e2879e0-1

**Comment by damencho** [comment]
Can you upload your js console logs from such a session?

**Comment by jusquiame** [comment]
Screen-sharing and JVB weird load start @ : Logger.js:154 2022-01-11T04:53:50.317Z 

[jitsiMeetP2PBandwidthLeak-jsConsole-1641876882835.log](https://github.com/jitsi/jitsi-meet/files/7843835/jitsiMeetP2PBandwidthLeak-jsConsole-1641876882835.log)

**Comment by damencho** [comment]
You are running jitsi-meet with modifications, right? Is ljm changed? 
What are the versions running?

**Comment by jusquiame** [comment]
Indeed I modified jitsi-meet, just a bit. 
Here are the little customization I made : 

- Added a LUA Script that hook the participants connections / disconnections , so that my Jitsi instances can notify my front server, for timestamping purpose. (done 1 year + ago, removing it changes nothing to described problem)

- Also added : `document.domain="myMainDomain.com";` at the beginning of app.bundle.min.js, for my main web app to be able and access the jitsi-meet iFrame. 

That's it. 

external-api is untouched, just the version shipped with configuration mentioned above :

jitsi-meet 2.0.6689-1,
prosody, tokens, turnserver, web, web-config 1.0.5638-1
videobridge2 2.1-592-g1e2879e0-1

Once again, this is a jitsi **+ Chrome** issue.

**Comment by jusquiame** [comment]
And many thanks for investigating on this. I use only Jitsi in a P2P context, and I find myself in a situation where an instance that was good for a thousand users or more can now break with only 30 for bandwidth reasons. As server costs run these days, that hurts a bit.

**Comment by jusquiame** [comment]
Hi, and thanks for the patch !
One last question : 
It's not packed in jan 17 2022 Jitsi version, or is it ? What version to install in order to test it ? Or when will it be in stable version ?

Thanks for your great job anyway. Jitsi is truly amazing.

**Comment by damencho** [comment]
> Or when will it be in stable version ?

No ETA as there was a stable 2 days ago.

This had been merged into lib-jitsi-meet, and hasn't reached jitsi-meet. Once that is done, you can get the version from the unstable debian repo or you can test it on alpha jitsi net or beta meet jit si.

**Comment by jusquiame** [comment]
Thanks !

**Comment by jusquiame** [comment]
It works !!! Bug fixed !!! Thanks again and again.
